Gcp/service/os_config_v2

Prem-CyberSec opened the pull request:
### Pull Request for the OS_Config_V2

**Resources used**
1. Policy Orchestrator
2. Policy Orchestrator for organization.

**Pending Resource**

1. Policy Orchestrator for Folder
_Reason: Terraform build is not supported due to multiple dependencies for the resource_


**NOTE: THIS BRANCH HAS BEEN REBASED MULTIPLE TIMES TO MAKE SURE THAT THE BRANCH IS UPTO DATE WITH THE LATEST HELPERS**

Prem-CyberSec committed:
Configured the branch for the OS-Config-V2 service

Prem-CyberSec committed:
Configured the branch for the OS-Config-V2 service in the policies directory

Prem-CyberSec committed:
Remove Unnecessary directories and files in my branch

Prem-CyberSec committed:
Added a policy for Lead to review

Need help in this policy, I can't get the OPA to evaluate this since it's outputing the error as undefined when I try to evaluate the Policy.

Prem-CyberSec committed:
Pushing my First Policy

This is my first policy that I am pushing to the repository. I will continue to work on more policy as we speak

Prem-CyberSec committed:
Updated README file

I have described the process to deploying the first policy in the README file

Prem-CyberSec committed:
Added new policy for attribute Action

I have added policy for action specifying that the if DELETE is passed in argument, it will be blacklisted.

Prem-CyberSec committed:
Added a new condition in the Orchestrated resource

A condition to verify the mode used by os-policies during orchestration is verified

Prem-CyberSec committed:
Added a policy to use correct archive type for APT repository

This addition ensures that we are only using safe and trusted sources if and when we are trying to install a new package

Prem-CyberSec committed:
Added a condition in action policy

A condition was set to check the state of the orchestration and depending on that along with the DELETE action it will be rejected if the state is active

Prem-CyberSec committed:
Added a policy which verifies based on labels and included_locations

This policy verifies that the orchestration scope can only be tested when certain labels and location is selected before deplyoment of the code.

Prem-CyberSec committed:
Added vars.rego for other resources

Prem-CyberSec committed:
Added Action and Orchestrated-Resource Policy

The Policies are similar to the ones we observed in the policy orchestrator with the main difference being that we assing a organization-id instead of a project-id and few additional blocks which are optional

Prem-CyberSec committed:
Updated the vars.rego for all of my resources

The update is in turn of using the new helpers which directly calls the name for the compliant and non-compliant resource

Prem-CyberSec committed:
Added a policy for pattern matching the ID

This policy takes a set pattern as an input from the provided options available to stick to the the specific tech stack the commpany might be using. This creates a homogeneous place to orchestrate the autonomous inputs.

Prem-CyberSec committed:
Added a policy for Organization ID

This policy shows a defined pattern that needs to be followed for defining the organization_id

Prem-CyberSec committed:
Removed resource- policy_orchestrator_for_folder

Removed this resource since it is not able to build on the terraform as the error says it does not support.

Prem-CyberSec committed:
Removed Files which got pulled during rebase

I had rebased my branch to get the new helpers update. During that time I got this issue that a new files were detected.

Seb-sketch commented on policies/gcp/os_config_v2/policy_orchestrator/action/policy.rego:
This logic is wrong. The policy description says "DELETE action is not allowed if state is active" but:

It blacklists state = "STOPPED" instead of checking for state = "ACTIVE"
The first condition part is fine but the 2nd has the same name "Check if the action is Delete" and should instead check for "state" being "ACTIVE" and should also be more accurately labeled

Seb-sketch commented on policies/gcp/os_config_v2/policy_orchestrator/orchestration_scope/policy.rego:
Both conditions have identical descriptions but check completely different attributes (locations vs environment labels). Please ensure each condition is accurately defined. The condition descriptions say "disallowed locations" but use whitelist policy, kind of contradictory language.

Seb-sketch commented on policies/gcp/os_config_v2/policy_orchestrator_for_organisation/action/policy.rego:
Same as previous action policy. This logic is wrong. The policy description says "DELETE action is not allowed if state is active" but:

It blacklists state = "STOPPED" instead of checking for state = "ACTIVE"
The first condition part is fine but the 2nd has the same name "Check if the action is Delete" and should instead check for "state" being "ACTIVE" and should also be more accurately labeled

Seb-sketch reviewed (changes requested):
Firstly you need to fix the deletion of the other files within the branch, these services from other students have been merged and if this PR is merged it will remove these files, you will need to bring these all back into your fork before this is merged.

Several package names use inconsistent naming conventions between files. 
All your policies and the vars.rego refer to “policy_orchestrator_for_organization”. But in your folders, you have it saved as .../gcp/os_config_v2/policy_orchestrator_for_organisation/ and thus they are not calling correctly. Ensure they all use the same spelling of organization.

Additionally the action policies have case sensitivity issues using Blacklist while others use blacklist, might not affect the outcome of testing but try to keep all the cases uniform for future automation processes.

The rest looks good.

Prem-CyberSec committed:
Addressed Naming Convention issue with Policy Orchestrator for Organization

Prem-CyberSec committed:
Removed policy_orchestrator_for_organisation

The issue has been attended to, under the directory name: policy_orchestrator_for_organization

Prem-CyberSec committed:
Addressed the Case Sensitivity Issue and Logic for action
